Features/docker server

crankycoder opened the pull request:
WIP:  

This adds a `server` directory to the project which should allow you to run a toy server instance that exposes the SGDClassifier.

You'll need to have at docker-ce to build the Docker image.

crankycoder committed:
Added virtual_env to gitignore directories

crankycoder committed:
Added boilerplate code of conduct, contributors and license files.

crankycoder committed:
init checkin of a server implementation and a single testcase

crankycoder committed:
Added a Dockerfile and scaffolding around getting a server instance up and running.

crankycoder committed:
added some more docs on the port numbers used by the sample service

BGluth commented:
I'm having an issue when running the container (just from running the `all` target in the Makefile). I think it's related to `mozfldp/simulation_util.py` being a symlink:

```bash
...
Step 10/11 : RUN python setup.py install
 ---> Running in cb1f95884d30
running install
running bdist_egg
running egg_info
creating mozilla_fldp.egg-info
writing mozilla_fldp.egg-info/PKG-INFO
writing dependency_links to mozilla_fldp.egg-info/dependency_links.txt
writing top-level names to mozilla_fldp.egg-info/top_level.txt
writing manifest file 'mozilla_fldp.egg-info/SOURCES.txt'
reading manifest file 'mozilla_fldp.egg-info/SOURCES.txt'
writing manifest file 'mozilla_fldp.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/mozfldp
copying mozfldp/server.py -> build/lib/mozfldp
error: can't copy 'mozfldp/simulation_util.py': doesn't exist or not a regular file
The command '/bin/sh -c python setup.py install' returned a non-zero code: 1
```
I'm not a docker guy, so it could very well also be that I'm doing something wrong.

crankycoder committed:
updated Makefile to copy the simulation_util.py file in before running
docker build

force any running container with 'mozfldp' to stop before a new one is
started

ignore the version of simulation_util.py that is copied in during the
build

BGluth commented:
That resolved it.

crankycoder commented:
Superceeded by #22
